Update Feature info and Gradle templates

matrei opened the pull request.

matrei committed:
fix: update feature names, descriptions and urls

matrei committed:
fix: update Gradle templates

matrei committed:
fix: set H2 as default feature

matrei committed:
fix: add title to the `Console` feature

matrei committed:
fix: update tests with new values

matrei committed:
fix: polish feature descriptions

jdaugherty commented on grails-forge/grails-forge-core/src/main/java/org/grails/forge/feature/database/H2.java:
```diff
@@ -20,12 +20,18 @@
 @Singleton
 @Primary
-public class H2 extends DatabaseDriverFeature {
+public class H2 extends DatabaseDriverFeature implements DefaultFeature {
```
H2 isn't required unless GORM is being used.  What's the thinking on adding this as a default feature?

matrei replied:
Good catch! I'll add another condition for when to apply it.

jdaugherty commented on grails-forge/grails-forge-core/src/main/java/org/grails/forge/feature/database/MongoGorm.java:
```diff
@@ -52,7 +52,7 @@ public String getTitle() {
     @Override
     public String getDescription() {
-        return "Configures GORM for MongoDB for Groovy applications";
+        return "Configure GORM for using MongoDB.";
```
What are your thoughts about changing GORM -> Grails Data since we've renamed the packages?

matrei replied:
Sounds good! But that is a big task.

jdaugherty replied:
We don't have to do it for this PR, let's discuss on the weekly?

jdaugherty commented on grails-forge/grails-forge-core/src/main/java/org/grails/forge/feature/database/Neo4jGorm.java:
```diff
@@ -44,7 +44,7 @@ public String getTitle() {
     @Override
     public String getDescription() {
-        return "Configures GORM for Neo4j for Groovy applications";
+        return "Configure GORM for using Neo4j";
```
Should we set a max supported version or something?  If they select Neo4j for Grails 7, it will fail ... seems like we should tell the user this isn't supported yet.

matrei replied:
Yes, we should probably disable it in some way.

jamesfredley replied:
Let's do this as a separate PR.  https://github.com/apache/grails-core/issues/15008

jdaugherty replied:
I 100% agree doing this separately, but wanted to point out we should create a ticket.  Thanks @jamesfredley

jdaugherty commented on grails-forge/grails-forge-core/src/main/java/org/grails/forge/feature/database/TestContainers.java:
```diff
@@ -138,6 +138,6 @@ public String getCategory() {
     @Override
     public String getThirdPartyDocumentation() {
-        return "https://www.testcontainers.org/";
+        return "https://www.testcontainers.org";
```
@jamesfredley changed the url to java.testcontainers.org instead of www.testcontainers.org in the docs, so should we change that here too?

jdaugherty commented on grails-forge/grails-forge-core/src/main/java/org/grails/forge/feature/grails/GrailsConsole.java:
```diff
@@ -41,9 +41,14 @@ public String getName() {
     @Override
     public String getDescription() {
-        return "Starts the Grails console, which is an extended version of the regular Groovy console.";
+        return "A Groovy console for interactive runtime application management and debugging.";
```
I don't think this is correct.  It's a grails console and it doesn't launch the interactive window.

matrei replied:
<img width="1111" height="627" alt="Skärmbild 2025-08-20 205629" src="https://github.com/user-attachments/assets/7ea3eee6-8104-4e41-9b77-31f2f8a4bced" />

jdaugherty replied:
I had no idea this was supported.  Thank you.  Leave it as is.

jdaugherty commented on grails-forge/grails-forge-core/src/main/java/org/grails/forge/feature/grailsWrapper/GrailsWrapper.java:
```diff
@@ -45,12 +45,12 @@ public String getName() {
     @Override
     public String getDescription() {
-        return "Gives you the ability to run grails shell from terminal.";
+        return "Enables running the Grails CLI directly without requiring a local Grails distribution installed.";
```
The wrapper is really a light weight installer for the delegating CLI.

matrei replied:
What would be a better description of the Grails Wrapper feature for a new user?

jdaugherty replied:
Gradle refers to their wrapper as: 

       The Wrapper is a script (called gradlew or gradlew.bat) that invokes a declared version of Gradle, downloading it beforehand if necessary. I

But ours downloads the version of grails matching the project version. 

     A tiny wrapper script that auto-downloads the Grails Delegating CLI on demand and transparently delegates all commands to it.

jdaugherty replied:
My main concern is that people understand it's a way to download the grails delegating CLI without checking in binaries.  It's meant to save space and be able to be checked in if not using sdkman

matrei replied:
>  is that people understand it's a way to download the grails delegating CLI without checking in binaries. It's meant to save space and be able to be checked in if not using sdkman

That is not how I see the `./grailsw` command with my end user hat on.
For me as an end user, it's a convenient shortcut to run/start the Grails CLI with the right version in a Grails project, plain and simple.

jamesfredley replied:
So it's 'what it does' vs 'how it works'.  Here is a stab at combining both ideas into a single sentence.

`A tiny wrapper script that enables running the Grails CLI directly by auto-downloading and transparently delegating commands.`

jdaugherty replied:
I'm ok with @jamesfredley 's suggestion.  Are you @matrei ?

matrei replied:
Perfect!

jdaugherty commented on grails-forge/grails-forge-core/src/main/java/org/grails/forge/feature/view/Scaffolding.java:
```diff
@@ -51,12 +51,14 @@ public String getName() {
     @Override
     public String getTitle() {
-        return "Grails Scaffolding Plugin";
+        return "Scaffolding";
```
We distinguish between libraries for logback, but not for plugins.  Shouldn't we be consistent?

matrei replied:
I started drifting into thinking in "Features", not implementations.
I don't know, it's not a always a clear cut.

jdaugherty replied:
Let's leave it unless @jamesfredley thinks otherwise?

jamesfredley replied:
I do not have a strong opinion.  Reviewing the changes it appears that 'plugin' was removed from the title for official apache grails plugins.  That aligns with the maven coordinate change and I think in general makes them appear as official parts of Grails, which could be a positive.

matrei committed:
fix: add h2 feature only when using gorm-hibernate

matrei committed:
fix: change testcontainers url

matrei committed:
Merge branch '7.0.x' into fork/matrei/update-features-and-templates

matrei committed:
Merge branch '7.0.x' into fork/matrei/update-features-and-templates

# Conflicts:
#	grails-forge/grails-forge-core/src/main/java/org/grails/forge/feature/database/TestContainers.java
#	grails-forge/grails-forge-core/src/main/java/org/grails/forge/feature/migration/DatabaseMigrationPlugin.java

jamesfredley committed:
Merge branch '7.0.x' into update-features-and-templates

# Conflicts:
#	grails-forge/grails-forge-core/src/main/java/org/grails/forge/feature/database/EmbeddedMongo.java
#	grails-forge/grails-forge-core/src/main/java/org/grails/forge/feature/database/MongoGorm.java

jamesfredley commented on grails-forge/grails-forge-core/src/main/java/org/grails/forge/feature/cache/EHCache.java:
```diff
@@ -55,12 +56,12 @@ public void apply(GeneratorContext generatorContext) {
     @Override
     public String getThirdPartyDocumentation() {
-        return "https://www.ehcache.org/";
+        return "https://www.ehcache.org";
```
I think we should leave trailing slashes, since without is technically missing the path portion of the URL.

jamesfredley commented on grails-forge/grails-forge-core/src/main/java/org/grails/forge/feature/cache/EHCache.java:
```diff
@@ -55,12 +56,12 @@ public void apply(GeneratorContext generatorContext) {
     @Override
     public String getDocumentation() {
-        return "https://grails-plugins.github.io/grails-cache-ehcache/latest/";
+        return "https://grails-plugins.github.io/grails-cache-ehcache/latest";
```
This causes two HTTP requests

<img width="1654" height="1480" alt="image" src="https://github.com/user-attachments/assets/e8f2896e-a14b-4d24-84dc-5e4b045442ac" />

jamesfredley commented on grails-forge/grails-forge-core/src/main/java/org/grails/forge/feature/cache/GrailsCache.java:
```diff
@@ -69,7 +70,7 @@ public String getCategory() {
     @Override
     public String getDocumentation() {
-        return "https://docs.grails.org/";
+        return "https://docs.grails.org/" + VersionInfo.getGrailsVersion() + "/guide/cache.html";
```
In the recent URL update PRs I ended up removing " + VersionInfo.getGrailsVersion() + " since this is invalid for SNAPSHOT versions.  I put /latest/ in these URLs.

jdaugherty replied:
This could cause confusion when there are multiple major versions though.  I assume the documentation is shown on the CLI somehow?  If so, I'd argue we need the version in all of these.  

We could easily publish the snapshot version in the docs too, that would fix the snapshot problem.  It might also makes it better since right now after publishing a 6.x snapshot, we would technically have to republish a 7.x.  I could build the same "latest" detection logic into the snapshot publishing that's there for release in the github action.

jamesfredley replied:
I have not seen where the documentation is exposed on start.grails.org or the forge cli.  There is a features schema on https://grailsforge-latest-cjmq3uyfcq-uc.a.run.app/swagger-ui/index.html#/, but documentation is not returned on endpoints like https://snapshot.grails.org/application-types/web/features

If we want to keep version here we should write another method in VersionInfo that returns 'SNAPSHOT' when the version end with -SNAPSHOT to be compatible with the documentation.

matrei replied:
I have implemented a `VersionInfo.getDocumentationVersion()` method in this PR that will take `snapshot` into account.
As @jdaugherty pointed out, in the best of worlds we would publish version specific snapshots and this method would return `6.2.4-snapshot` and `7.0.0-snapshot`, but perhaps we can ticket that if we want that functionality.

jamesfredley replied:
Sounds great, thank you.

jamesfredley commented on grails-forge/grails-forge-core/src/main/java/org/grails/forge/feature/database/MongoSync.java:
```diff
@@ -71,7 +71,7 @@ public String getCategory() {
     @Override
     public String getDocumentation() {
-        return "https://docs.mongodb.com/drivers/java/sync/current/";
+        return "https://docs.mongodb.com/drivers/java/sync/current";
```
two http requests

jamesfredley commented on grails-forge/grails-forge-core/src/main/java/org/grails/forge/feature/database/TestContainers.java:
```diff
@@ -138,6 +138,6 @@ public String getCategory() {
     @Override
     public String getThirdPartyDocumentation() {
-        return "https://java.testcontainers.org/";
+        return "https://java.testcontainers.org";
```
missing path

jamesfredley commented on grails-forge/grails-forge-core/src/main/java/org/grails/forge/feature/migration/DatabaseMigrationPlugin.java:
```diff
@@ -35,21 +36,22 @@ public String getName() {
 
     public String getThirdPartyDocumentation() {
-        return "https://www.liquibase.org/";
+        return "https://www.liquibase.org";
```
missing path

jamesfredley commented on grails-forge/grails-forge-core/src/main/java/org/grails/forge/feature/other/HibernateValidator.java:
```diff
@@ -63,6 +63,6 @@ public String getCategory() {
     @Override
     public String getDocumentation() {
-        return "https://hibernate.org/validator/";
+        return "https://hibernate.org/validator";
```
two http requests

jamesfredley commented on grails-forge/grails-forge-core/src/test/groovy/org/grails/forge/feature/database/MongoSyncSpec.groovy:
```diff
@@ -33,7 +33,7 @@ class MongoSyncSpec extends ApplicationContextSpec implements CommandOutputFixtu
         then:
         readme
-        readme.contains("https://docs.mongodb.com/drivers/java/sync/current/")
+        readme.contains("https://docs.mongodb.com/drivers/java/sync/current")
```
https://www.mongodb.com/docs/drivers/java/sync/current/ is the new full URL

jamesfredley reviewed (commented):
I provided feedback on the open comment discussions and added some comments on the URLs.  URLs are top of mind after https://github.com/apache/grails-core/pull/14988 and https://github.com/apache/grails-core/pull/15003

matrei committed:
fix: feature documentation urls

matrei committed:
fix: use the correct version for feature documentation links

Add a method `VersionInfo.getDocumentationVersion()` that
returns the string `snapshot` or the Grails version and
use that in the feature documentation links.

matrei committed:
fix: improve the Grails Wrapper description

matrei commented:
Hmm, that's a strange test failure in the Groovy Joint Validation Build: https://github.com/apache/grails-core/actions/runs/17173238880/job/48725756373?pr=14994#step:9:11029

jamesfredley commented:
@matrei It was something in the GitHub action cache.
